src/components/Editor/SearchBar.js

```diff
@@ -213,26 +221,28 @@ const SearchBar = React.createClass({
     return findDOMNode(this).querySelector("input");
   },
 
-  updateFunctionSearchResults(query: string) {
+  updateSymbolSearchResults(query: string) {
     const {
       sourceText,
       updateSearchResults
     } = this.props;
+    const { selectedSymbolType } = this.state;
 
     if (query == "" || !sourceText) {
       return;
     }
 
-    const functionDeclarations = getSymbols(sourceText.toJS()).functions;
+    const symbolDeclarations = getSymbols(
```

Comment:
This function `updateSymbolSearchResults` is called whenever `doSearch` is called so we're parsing the document every time we search. I think we should be able to move the `symbolDeclarations` either to state or props so they're parsed out only once per document.